installed-tests/js/testGObjectInterface.js

```diff
@@ -0,0 +1,323 @@
+// -*- mode: js; indent-tabs-mode: nil -*-
+
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const GObject = imports.gi.GObject;
+const JSUnit = imports.jsUnit;
+const Lang = imports.lang;
+const Mainloop = imports.mainloop;
+
+const MyInterface = new Lang.Interface({
+    Name: 'MyInterface',
+});
+
+const MyInterfaceGObject = new Lang.Class({
+    Name: 'MyInterfaceGObject',
+    Extends: GObject.Object,
+    Implements: [ MyInterface ],
+
+    _init: function (props={}) {
+        this.parent(props);
+    }
+});
+
+const MyGObjectInterface = new Lang.Interface({
+    Name: 'MyGObjectInterface',
+    GTypeName: 'ArbitraryGTypeName',
+    Requires: [ GObject.Object ],
+    Properties: {
+        'interface-prop': GObject.ParamSpec.string('interface-prop',
+            'Interface property', 'Must be overridden in implementation',
+            GObject.ParamFlags.READABLE,
+            'foobar')
+    },
+    Signals: {
+        'interface-signal': {}
+    },
+
+    requiredG: Lang.Interface.UNIMPLEMENTED,
+    optionalG: function () {
```

Comment:
Hm okay, interesting. I've always used `GInterface` in a way where no interface function has a default implementation, but as long as its consistent with the way it works there then I guess that makes sense.